conans/client/generators/visualstudio.py

```diff
@@ -116,4 +117,29 @@ def content(self):
             userprofile = userprofile.replace("\\", "\\\\")
             formatted_template = re.sub(userprofile, "$(USERPROFILE)", formatted_template,
                                         flags=re.I)
+
+        if self.conanfile.name or self.conanfile.version:
```

Comment:
I think maybe it is better to add this to the template instead of adding them dynamically. 

I am fine if they are an empty string if not defined, it is typically easier to check an empty string than a missing variable.

Also, the casing of the variables better be aligned with the existing ones:
```xml
<ConanCompilerFlags>{compiler_flags}</ConanCompilerFlags>
    <ConanLinkerFlags>{linker_flags}</ConanLinkerFlags>
    <ConanPreprocessorDefinitions>{definitions}</ConanPreprocessorDefinitions>
```

So probably ``ConanPackageName``, ``ConanPackageVersion``

Reply:
Consider adding a test also if possible, to validate the new functionality is there and it is not broken in future changes.

Reply:
> I think maybe it is better to add this to the template instead of adding them dynamically.

Ok, Agreed. 

> I am fine if they are an empty string if not defined, it is typically easier to check an empty string than a missing variable.

That was my hesitancy. And I didn't really want to tackle turning this into a Jinga template -- *not sure why that isn't being used here*.  The `visualstudio_multi` generator was using `minidom` so I figured I'd roll the dice ... but was sort of expecting it to get shot down :)

Reply:
We try to use the templates whenever possible, and only rely on dom modification when it is really necessary, like the "multi" functionality that needs to insert new nodes, one per install.

Reply:
> Consider adding a test also if possible, to validate the new functionality is there and it is not broken in future changes. Ask for help if you need, or we can also add them later.

I've added a test ... but am struggling to get the test suite to run. Need to ask @puetzk when he gets in.

Reply:
Looking at the `msbuild` generator, I'm guessing I'd want to add `ConanPackageName` and `ConanPackageVersion` here:
https://github.com/conan-io/conan/blob/0437d7a516213b011f70d27fd314652d8fca8f43/conans/client/generators/msbuild.py#L13-L19

But I'm not yet sure where's the appropriate spot to perform the substitution.

Reply:
Think I have a working `msbuild` integration in https://github.com/conan-io/conan/pull/7645/commits/1f998f478e7a9da812169112d3b145cba590077b